hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/ConsumerMessageSender.java

```diff
@@ -142,13 +142,19 @@ public void accept(MessageSendingResult result) {
                 handleFailedSending(message, result);
                 if (!isTtlExceeded(message) && shouldRetrySending(result)) {
                     retrySingleThreadExecutor.schedule(() -> retrySending(result),
-                            subscription.getSubscriptionPolicy().getMessageBackoff(), TimeUnit.MILLISECONDS);
+                            extractRetryDelay(result), TimeUnit.MILLISECONDS);
```

Comment:
Putting jobs with delay equal to Integer.MAX will cause them to stay for... Integer.MAX milliseconds in our queue...